### PR TITLE
[cxx-interop] Emit IR for destructors of temporary C++ values

### PR DESCRIPTION
**Pull request by egorzhdan**
This fixes the "undefined reference" linker errors.

rdar://101092732

### CODE REVIEW

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by hyp** [comment]
THanks!

**Review by hyp — APPROVED** [review]

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Windows

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Windows

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

File: test/Interop/Cxx/class/destructors-with-temporary-values-irgen.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// RUN: %target-swiftxx-frontend -emit-ir -I %S/Inputs -validate-tbd-against-ir=none %s | %FileCheck %s
+
+// TODO: make this test work on Windows
+// XFAIL: OS=windows-msvc
```
Comment (hyp) [inline test/Interop/Cxx/class/destructors-with-temporary-values-irgen.swift:4]: why can't you add a match line for Windows that matches the right function name ABI? We should stop proliferating XFAIL windows in tests if there's no real need for it.

Comment (egorzhdan) [reply test/Interop/Cxx/class/destructors-with-temporary-values-irgen.swift:4]: I couldn't get the Windows name out of the CI, and I don't have a Windows machine to test this locally. If this fix is not super urgent, I'll find a Windows VM to run this.

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Windows

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Windows

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Windows

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Windows

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Windows

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Pull request merged**